### PR TITLE
Update default Xcode to 15.0.

### PR DESCRIPTION
**Pull request by brandonpage**
I will additional Xcode 14 builds for iOS 15.5 testing via CircleCI triggered pipeline (and remove the current xcode 15 beta one) after this is merged.

### CODE REVIEW

**Review by bbirman — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -2,10 +2,10 @@
...
 #  NOTE: When updaing Xcode check the manifest for compatible chruby versions.
 anchors:
-  - &android-env     "cimg/android:2023.08.1-node"
-  - &latest-xcode    "14.3.1"
-  - &latest-ios      "16.4"
-  - &min-ios         "15.5"
+  - &android-env     "cimg/android:2023.09.1-node"
+  - &latest-xcode    "15.0.0"
+  - &latest-ios      "17.0"
+  - &min-ios         "16.4"
```
Comment (wmathurin) [inline .circleci/config.yml:8]: Should the min ios go up now or in 12.0 ?

Comment (brandonpage) [reply .circleci/config.yml:8]: The min will go up in 12.  The issue is that the images only contain have 2 iOS versions.   

Before:
- xcode 14 default for iOS 15.5 and 16.4 testing
- xcode 15 additional build for iOS 17 testing

After:
- xcode 15 default for iOS 16.4 and 17.0 testing
- xcode 14 additional build for 15.5 testing  <- we can drop this without code changes when MSDK 12 goes out.

**Comment by brandonpage** [comment]
I am struggling to get Xcode 15 tests working (despite local tests being fine) because Apple has royally messed up their virtualized macOS environment (the Apple Virtualized Framework) with this latest release.  [Many other users are suffering from abysmal performance, crashes and simulators simply not working.](https://discuss.circleci.com/t/severe-performance-problems-with-xcode-15/49205/29)  I will continue to look for a fix/workaround, but I am not sure one exists.

File: .circleci/config.yml (hunk 1 above)
Comment (wmathurin) [reply .circleci/config.yml:8]: 👍

**Pull request closed without merging**